subprojects/core/src/integTest/groovy/org/gradle/api/tasks/bundling/ZipIntegrationTest.groovy

```diff
@@ -40,10 +40,9 @@ class ZipIntegrationTest extends AbstractIntegrationSpec {
             }
         '''
 
-        when:
-        fails 'zip'
-        then:
-        failure.assertHasCause('Encountered duplicate path "file1.txt" during copy operation configured with DuplicatesStrategy.FAIL')
+        expect:
+        executer.expectDeprecationWarning('Encountered duplicate path during copy operation configured with default duplicates strategy. Allowing duplicates with the default duplicates strategy has been deprecated. This is scheduled to be removed in Gradle 7.0. Duplicate path: "file1.txt". Explicitly set the duplicates strategy to \'DuplicatesStrategy.INCLUDE\' if you want to allow duplicate paths.')
```

Comment:
> Allowing duplicates with the default duplicates strategy has been deprecated.

This sounds contrived. I don't think it adds much to the (already long) message, so WDYT about removing it?

Reply:
I'd move mentioning the duplicate entry towards the beginning of the message.

Reply:
The deprecation message needs to contain something with `... has been deprecated`. Moreover, I wanted to have the file which caused the duplication as context, so the given deprecations count as the same. Do you agree with those constraints? Do you have a suggestion to make the message better having those constraints in mind?